src/modules/cpu/kernel/slice.hpp

```diff
@@ -26,184 +26,213 @@ SOFTWARE.
+                            T *dstPtr,
+                            RpptGenericDescPtr dstGenericDescPtr,
+                            Rpp32s *anchorTensor,
+                            Rpp32s *shapeTensor,
+                            T* fillValue,
+                            bool enablePadding,
+                            Rpp32u *roiTensor,
+                            RppLayoutParams layoutParams,
+                            rpp::Handle& handle)
 {
-    RpptROI3D roiDefault;
-    if(srcGenericDescPtr->layout==RpptLayout::NCDHW)
-        roiDefault = {0, 0, 0, (Rpp32s)srcGenericDescPtr->dims[4], (Rpp32s)srcGenericDescPtr->dims[3], (Rpp32s)srcGenericDescPtr->dims[2]};
-    else if(srcGenericDescPtr->layout==RpptLayout::NDHWC)
-        roiDefault = {0, 0, 0, (Rpp32s)srcGenericDescPtr->dims[3], (Rpp32s)srcGenericDescPtr->dims[2], (Rpp32s)srcGenericDescPtr->dims[1]};
     Rpp32u numThreads = handle.GetNumThreads();
+    Rpp32u numDims = srcGenericDescPtr->numDims - 1; // exclude batchsize from input dims
 
     omp_set_dynamic(0);
 #pragma omp parallel for num_threads(numThreads)
     for(int batchCount = 0; batchCount < dstGenericDescPtr->dims[0]; batchCount++)
     {
-        RpptROI3D roi;
-        RpptROI3DPtr roiPtrInput = &roiGenericPtrSrc[batchCount];
-        compute_roi3D_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+        T *srcPtrTemp, *dstPtrTemp;
+        srcPtrTemp = srcPtr + batchCount * srcGenericDescPtr->strides[0];
+        dstPtrTemp = dstPtr + batchCount * dstGenericDescPtr->strides[0];
 
-        Rpp32f *srcPtrImage, *dstPtrImage;
-        srcPtrImage = srcPtr + batchCount * srcGenericDescPtr->strides[0];
-        dstPtrImage = dstPtr + batchCount * dstGenericDescPtr->strides[0];
+        T *srcPtrChannel, *dstPtrChannel;
+        dstPtrChannel = dstPtrTemp;
 
-        Rpp32u bufferLength = roi.xyzwhdROI.roiWidth * layoutParams.bufferMultiplier;
+        Rpp32s *anchor = &anchorTensor[batchCount * numDims];
+        Rpp32s *shape = &shapeTensor[batchCount * numDims];
 
-        Rpp32f *srcPtrChannel, *dstPtrChannel;
-        dstPtrChannel = dstPtrImage;
+        // get the starting address of length values from roiTensor
+        Rpp32u *roi = roiTensor + batchCount * numDims * 2;
+        Rpp32s *length = reinterpret_cast<Rpp32s *>(&roi[numDims]);
 
-        // Slice without fused output-layout toggle (NCDHW -> NCDHW)
-        if((srcGenericDescPtr->layout == RpptLayout::NCDHW) && (dstGenericDescPtr->layout == RpptLayout::NCDHW))
+        if (numDims == 4)
         {
-            srcPtrChannel = srcPtrImage + (roi.xyzwhdROI.xyz.z * srcGenericDescPtr->strides[2]) + (roi.xyzwhdROI.xyz.y * srcGenericDescPtr->strides[3]) + (roi.xyzwhdROI.xyz.x * layoutParams.bufferMultiplier);
-
-            Rpp32u copyLengthInBytes = bufferLength * sizeof(Rpp32f);
-            for(int c = 0; c < layoutParams.channelParam; c++)
+            // order of dims
+            Rpp32s dimsOrder[3];
```

Comment:
please initialize in one line dimsOrder[3] = NCDHW? {1, 2, 3} : {0, 1, 2}

Reply:
Not in the PR